File: Go/Easy/Middle Node/solution_1.go
package main

type LinkedList struct {
	Value int
	Next  *LinkedList
}

func MiddleNode(linkedList *LinkedList) *LinkedList {
	currentNode := linkedList
	count := 0

	for currentNode != nil {
		currentNode = currentNode.Next
		count += 1
	}

	middleNode := linkedList

	for i := 0; i < (count / 2); i++ {
		middleNode = middleNode.Next
	}

	return middleNode
}

File: Python/Easy/Middle Node/solution_1.py
class LinkedList:
    def __init__(self,value):
        self.value = value
        self.next = None

def middleNode(linkedList):
    count = 0
    currentNode = linkedList

    while currentNode is not None:
        currentNode = currentNode.next
        count += 1
    
    middleNode = linkedList

    for _ in range(count // 2):
        middleNode = middleNode.next
    
    return middleNode
    
    


